mlflow/utils/uri.py

```diff
@@ -281,3 +287,45 @@ def dbfs_hdfs_uri_to_fuse_path(dbfs_uri):
         )
 
     return _DBFS_FUSE_PREFIX + dbfs_uri[len(_DBFS_HDFS_URI_PREFIX) :]
+
+
+def resolve_uri_if_local(local_uri):
+    """
+    if `local_uri` is passed in as a relative local path, this function
+    resolves it to absolute path relative to current working directory.
+
+    :param local_uri: Relative or absolute path or local file uri
+
+    :return: a fully-formed absolute uri path or an absolute filesystem path
+    """
+    from mlflow.utils.file_utils import local_file_uri_to_path
+
+    if local_uri is not None and is_local_uri(local_uri):
+        scheme = get_uri_scheme(local_uri)
+        cwd = pathlib.Path.cwd()
+        local_path = local_file_uri_to_path(local_uri)
+        if not pathlib.Path(local_path).is_absolute():
+            if scheme == "":
+                if platform.system().lower() == "windows":
+                    return urllib.parse.urlunsplit(
+                        (
+                            "file",
+                            None,
+                            cwd.joinpath(local_path).as_posix(),
+                            None,
+                            None,
+                        )
+                    )
+                return cwd.joinpath(local_path).as_posix()
+            local_uri_split = urllib.parse.urlsplit(local_uri)
+            resolved_absolute_uri = urllib.parse.urlunsplit(
+                (
+                    local_uri_split.scheme,
+                    None,
+                    cwd.joinpath(local_path).as_posix(),
+                    local_uri_split.query,
+                    local_uri_split.fragment,
```

Comment:
Do we need the if / else conditional for Windows here, or could we consolidate the logic? Seems like we can

Reply:
I believe we need the if/else somewhere because we need a file URI for windows and a local path for Unix systems in this logic handling

Reply:
hmmm, actually i take that back :)
I think we can consolidate this

Reply:
Ok, I think the current version of this method is what I can do best without convoluting that logic further. if you have any better idea, I am happy to amend it.